CardGame.playground/Contents.swift

```diff
@@ -144,5 +202,110 @@ import Foundation
 //: Player 1 drew a 2 of diamonds, player 2 drew a ace of diamonds.
 //: Player 1 wins with 2 of diamonds.
 //: ```
-
-
+let tracker = CardGameTracker()
+let game = HighLow()
+game.delegate = tracker
+
+game.play()
+
+
+print("")
+print("")
+
+
+
+// Just for fun - Here's a different version
+
+// High Low - The Extended Version
+// As the cards are drawn, they are removed from the deck.
+// Scores are kept - 1 point awarded to winner of each round. (No points awarded if it's a tie.)
+// Game is over when all the cards have been drawn (26 rounds). Winner is announced based on final score.
+
+extension Deck {
+    mutating func drawCard2() -> Card {
+        let cardIndex = Int.random(in: 0..<self.cards.count)
+        let card = self.cards[cardIndex]
+        self.cards.remove(at: cardIndex)
+        return card
+    }
+}
+
+extension CardGameDelegate {
+    func gameDidEnd(player1Score: Int, player2Score: Int) {
+        if player1Score == player2Score {
+            print("The players have tied with a final score of \(player1Score)!")
+        } else if player1Score > player2Score {
+            print("Player 1 has won the game with a final score of \(player1Score).")
+        } else {
+            print("Player 2 has won the game with a final score of \(player2Score).")
+        }
+    }
+}
+
+class CardGameTracker2: CardGameDelegate {
+    func gameDidStart(_ game: CardGame) {
+        if game is HighLow2 {
+            print("Started a new game of High Low - Extended Version.")
+        }
+    }
+    
+    func game(player1DidDraw card1: Card, player2DidDraw card2: Card) {
+        print("Player 1 drew the \(card1) and Player 2 drew the \(card2).")
+    }
+    
+    func gameDidEnd(player1Score: Int, player2Score: Int) {
+        if player1Score == player2Score {
+            print("The players have tied with a final score of \(player1Score).")
+        } else if player1Score > player2Score {
+            print("Player 1 has won the game with a final score of \(player1Score).")
+        } else {
+            print("Player 2 has won the game with a final score of \(player2Score).")
+        }
+    }
+}
+
+class HighLow2: CardGame {
+    var deck = Deck()
+    var delegate: CardGameDelegate?
+    
+    func play() {
+        delegate?.gameDidStart(self)
+        
+        var player1Score = 0
+        var player2Score = 0
+        
+        var player1Card: Card
+        var player2Card: Card
+        
+        for _ in 1...26 {
```

Comment:
Nothing wrong with doing it this way, but if the size of your deck changed, you would need to maintain this number. You could derive the 26 from the decks count or just use a while loop to keep dealing cards as long as there are cards to deal:

``` swift
while deck.cards.count >= 2
```